src/_pytest/capture.py

```diff
@@ -621,10 +568,10 @@ class FDCapture(FDCaptureBinary):
     EMPTY_BUFFER = str()  # type: ignore
 
     def snap(self):
-        res = super().snap()
```

Comment:
Since `tmpfile` is now text-mode for reading also (not just for writing like the weird `EncodedFile` did), I switched to read directly. Also symmetric with the `Sys` variants which do the same.